src/foam/nanos/menu/Menu.js

```diff
@@ -70,13 +82,77 @@ foam.CLASS({
       name: 'icon',
       documentation: 'Icon associated to the menu item.',
       displayWidth: 80
+    },
+    {
+      documentation: 'Predicate providing arbitrary checks, in addition to the regular menu auth checks.',
+      class: 'foam.mlang.predicate.PredicateProperty',
+      name: 'readPredicate',
+      view: {
+        class: 'foam.u2.view.JSONTextView'
+      },
+      javaFactory: `
+      return foam.mlang.MLang.TRUE;
+      `,
     }
   ],
 
   methods: [
     function launch_(X, e) {
       this.lastMenuLaunchedListener && this.lastMenuLaunchedListener(this);
       this.handler && this.handler.launch(X, this, e);
+    },
+    {
+      documentation: 'Desire to call read predicate with calling context but predicate may also need access to this menu; add the current menu as context key MENU',
+      name: 'f',
+      type: 'Boolean',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        }
+      ],
+      javaCode: `
+      return getReadPredicate().f(
```

Comment:
[**Optional**]

Personally, I'd prefer that we indent javaCode one level beyond the `javaCode:` line to visually differentiate it from the property properties. Right now if I scan down the left side to see the property property keys, I see the following:

* documentation
* name
* type
* args
* javaCode
* return
* )

Obviously the last two don't belong because they're part of the value, not keys. Just my opinion though.

Reply:
I'm been the other way with indenting so that the resulting generated java looks correct, but as you prefer the javascript/java to be more readable.  We should add some standard/guide to our wiki

Reply:
All formatting, spacing issues addressed.